### ISSUE TITLE
Action continues when appname-slotname mismatches SCM credentials

### ISSUE THREAD

**Issue by Hazhzeng** [Azure/functions-action#25]
The function app will still be deployed to production when using SCM credentials even if the slot-name is set. Due to the limitation of the SCM credentials, the GitHub Action only have access to 1 site (either production or slot).
```
    - name: 'Run Azure Functions Action'
      uses: Azure/functions-action@v1
      id: fa
      with:
        app-name: hazeng-fa-node-stage
        package: '.'
        publish-profile: ${{ secrets.SCM_CREDENTIALS }}
        slot-name: this-will-be-ignored
```
There's a way to mitigate this, that is to add a dash with slotname in the app-name, and use SCM_CREDENTIALS of the slot for deployment.
```
    - name: 'Run Azure Functions Action'
      uses: Azure/functions-action@v1
      id: fa
      with:
        app-name: hazeng-fa-node-stage-slotnamehere
        package: '.'
        publish-profile: ${{ secrets.SCM_CREDENTIALS_OF_THE_SLOT }}
```

**Comment by Hazhzeng** [comment]
Discussed with @ahmedelnably, we will honor the endpoint in SCM_CREDENTIALS when publishing to a slot.
If the endpoint in SCM_CREDENTIALS does not match **appname-slotname**, we will throw a warning message.

**Comment by ahmedelnably** [comment]
Actually I'd prefer if we through an error in the case of a mismatch

**Comment by Hazhzeng** [comment]
currently, the action will honor the endpoint in SCM_CREDENTIALS when publishing (slot/production).

As mentioned above, we should error out if the user provides a mismatched **appname-slotname**

**Comment by Hazhzeng** [comment]
This is fixed in v1.1.3 release.
